Bump Sentry Gradle Plugin to 4.11.0

wzieba opened the pull request:
To address:
- https://github.com/wordpress-mobile/WordPress-Android/security/dependabot/77

Sentry Gradle Plugin in version 4.3.1 [bundled Gradle API](https://mvnrepository.com/artifact/io.sentry/sentry-android-gradle-plugin/4.3.1), which included old transitive dependencies with some security vulnerabilities.

### Testing
Not needed: the green light on CI is fine, locally I run the upload of mapping files, and it all worked well:

<img width="1505" alt="Screenshot 2024-08-09 at 12 58 02" src="https://github.com/user-attachments/assets/2b1216f6-b841-4368-b5cb-107d498fff5c">

wzieba committed:
Bump Sentry Gradle Plugin to 4.11.0

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

wzieba committed:
Submit dependencies to GH graph for every push

This can help in making sure that given change addresses (or introduces!) a security alert

wzieba commented on .github/workflows/submit-gradle-dependencies.yml:
```diff
@@ -1,9 +1,5 @@
-    branches:
-      - trunk
-      - release/*
+on: [push, pull_request]
```
I expected that with this change we might get 
> This pull request would resolve a Dependabot alert on <dependency>

message like e.g. here: https://github.com/wordpress-mobile/WordPress-Android/pull/19492

but it seems it doesn't work like this. Reverting this change.

wzieba replied:
I've asked a question here: https://github.com/gradle/github-dependency-graph-gradle-plugin/issues/154

ParaskP7 replied:
Someone actually responded to you on that! 👍

wzieba committed:
Revert "Submit dependencies to GH graph for every push"

This reverts commit 74dbfae7b04465aaef385eda465b92d0b4c67085.

ParaskP7 reviewed (approved):
Thanks for this update @wzieba , approved! 👍 

> Not needed: the green light on CI is fine, locally I run the upload of mapping files, and it all worked well:

Are you sure that such changes not need testing, and 🟢 CI is enough, I am just thinking future here and what I (or any product engineer) would have done to make sure such an update is not breaking at least the major Sentry flows (like reporting a crash)? 🤔

For example, would you consider adding a (breaking) commit (and then reverting it) to be a way to test for a crash reported. And then, asking from the reviewer to test on their side as well, by checking-out this specific (breaking) commit? Would that be too much you think? 🤔

wzieba commented:
The Sentry Gradle Plugin shouldn't influence how crashes are reported - we use it only for sending debug files (mappings and source contexts) and for manipulation to get database performance tracing. 

We don't change any runtime code here, hence no dependency diff on the runtime configuration reported. So I think it's totally safe to proceed with this bump without additional testing, WDYT?

ParaskP7 commented:
👋 @wzieba !

> The Sentry Gradle Plugin shouldn't influence how crashes are reported - we use it only for sending debug files (mappings and source contexts) and for manipulation to get database performance tracing.

True, I always perplex Sentry with Tracks and Tracks with Sentry, in the sense that I am very aware that Tracks is using Sentry extensively and me being cautious that updating Sentry, even if that's the plugin (not a dependency), it could somehow influence Sentry reporting crashes. 🤷 

To your (Sentry) understanding, this is not possible, right? 🤔 

> We don't change any runtime code here, hence no dependency diff on the runtime configuration reported.

I think we never report dependency diff for plugins anyway, right? 🤔 

> So I think it's totally safe to proceed with this bump without additional testing, WDYT?

Yea, I guess so, I am just trying to figure out what's the correct testing process for a Sentry bump, apologies for the noise... 🫣

wzieba commented:
> To your (Sentry) understanding, this is not possible, right? 🤔

In this context - I don't see plugin affecting how Sentry reports at runtime.

> I think we never report dependency diff for plugins anyway, right? 🤔

You're right, we don't. I meant changing runtime code here (in opposite to the build code). Plugins still can influence runtime dependencies, in case of Sentry it could've done that with `autoInstalltion.enabled = true`. So the lack of diff report of runtime dependencies proves that plugin update didn't update the Sentry SDK we use at runtime.

> Yea, I guess so, I am just trying to figure out what's the correct testing process for a Sentry bump, apologies for the noise... 🫣

No worries! It was a good question, and it's good to look for the optimal strategy for what's worth testing.

ParaskP7 commented:
Thanks for all the above clarification @wzieba ! 🙇
